### ISSUE TITLE
Current param not being respected

### ISSUE THREAD

**Issue by beltranrengifo** [hipstersmoothie/storybook-dark-mode#206]
Hi there!
I'm having a bug with this storybook [here](https://apps.mews.com/mews-ui-storybook/prerelease/RND-128890/0.0.0-RND-128890.5/index.html).
I set the `current` param as `light` but I can see how it changes when color schema in the system preferences is changed. Also, if OS set to dark mode and the SB is opened in a fresh window, the SB is shown in dark mode... :(

As stated in this [pr](https://github.com/hipstersmoothie/storybook-dark-mode/pull/98/commits/234a372e333e6a77e78888faab606bc9ab14ab6c) and in the documentation, the other way is expected, isn't it?

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/19485114/208877670-55bf7e4b-093c-4637-afa3-5c922db7dd97.png">
<img width="1503" alt="image" src="https://user-images.githubusercontent.com/19485114/208877694-0ccb1ede-56c3-458b-ae4a-42f110e2d833.png">

```
export const parameters = {
    docs: {
        container: DocsContainer,
        page: DocsPage,
        inlineStories: true,
        prepareForInline: (Story: StoryType) => <Story />,
    },
    layout: 'centered',
    options: {
        storySort: {
            method: 'alphabetical',
            order: STORYBOOK_ROOTS_ORDER,
        },
    },
    darkMode: {
        dark: { ...themes.dark },
        light: { ...themes.normal },
        current: 'light',
        stylePreview: true,
    },
};
```

Is this a bug or am I missing something?
Thanks!

**Comment by beltranrengifo** [comment]
hi!
after digging a bit, in the case that a user sets `current: 'light'` and the OS in dark mode, my guess is:
- on first render we have no `initialMode`
- `defaultMode` is still `undefined`
- but `prefersDark.matches` so `updateMode('dark')` is triggered
- the store is created with the `dark` value
- from here, [this line](https://github.com/hipstersmoothie/storybook-dark-mode/blob/5696ac7839d106a29d70abe6ea5c75b2cc9fd3ee/src/Tool.tsx#L222) prevents `current` from working, I think

I guess the problem is around [this ifs](https://github.com/hipstersmoothie/storybook-dark-mode/blob/5696ac7839d106a29d70abe6ea5c75b2cc9fd3ee/src/Tool.tsx#L222)
I'll try to take a closer look

**Comment by beltranrengifo** [comment]
Fixed in [release v2.0.5](https://github.com/hipstersmoothie/storybook-dark-mode/releases/tag/)
